files/scripts/dpkg-state

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python
+import sys
+import subprocess
+import signal
+
+# Get dpkg status.
+# Errors are printed to stderr, and should cause None to be returned.
+def dpkg_state(package):
+	state = 'not-installed'
+
+	# As of Debian 8, this is the only documented method.
+	# http://unix.stackexchange.com/questions/325594/script-a-test-for-installed-debian-package-error-handling/325595#325595
+
+	command = ['dpkg-query', '--show',
+	           '--showformat=${Package} ${Status}\n']
+
+	proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+	for line in proc.stdout:
+		line = line.decode('ASCII')
+		line = line.rstrip()
+		# "package desired-action error-flags state"
+		(p, _, _, s) = line.split(' ')
+		if p == package:
+			state = s
+			break
+	proc.stdout.close()
+
+	# Process should either have returned 0,
+	# or been killed because we closed the pipe (stdout).
+	# Otherwise, we had an error.
+	proc.wait()
+	if proc.returncode not in [0, -signal.SIGPIPE]:
+		return None
+
+	return state
```

Comment:
minor nitpick: state may not necessarily be initialised if proc.stdout does not contain the line of desire.  How about to raise in the else clause of the for loop?